### PR TITLE
feat(grep): move cypress-grep to @cypress/grep

### LINKED ISSUE #140: Moving to Core Cypress Repo
In an effort to better maintain and enhance this official plugin, we will be moving it into the [main Cypress repo](https://github.com/cypress-io/cypress) sometime in the next few weeks. During that time, we will be going through issues and PRs to make sure all relevant items are moved over and labeled appropriately.

Once it is moved, this repo will be archived and will become read-only.

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes https://github.com/cypress-io/cypress-grep/issues/140

### User facing changelog
Moves `cypress-grep` plugin into `@cypress/grep`

### Additional details
In an effort to better maintain and enhance this official plugin, we will be moving it into the main Cypress repo.

### Steps to test
`yarn workspace @cypress/grep test`

### PR Tasks

- [X] Have tests been added/updated?
- [X] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [X] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/grep/cypress/e2e/omit-and-skip-spec.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/// <reference path="../../src/index.d.ts" />
+
+// @ts-check
+
```
Comment (emilyrohrbough) [inline npm/grep/cypress/e2e/omit-and-skip-spec.js]:
```suggestion
```

File: npm/grep/cypress/e2e/tags/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Testing tags
+
+Test case for [#92](https://github.com/cypress-io/cypress-grep/issues/92)
```
Comment (emilyrohrbough) [inline npm/grep/cypress/e2e/tags/README.md]: Still relevent? Should we transfer issue to this repo?

File: npm/grep/expects/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Different JSON files with expected test statuses when running Cypress tests with different grep argument.
+
+Used via [cypress-expect](https://github.com/bahmutov/cypress-expect) with `--expect` option to run on CI to verify the correct tests were pending or passing.
```
Comment (emilyrohrbough) [inline npm/grep/expects/README.md:3]: We should log a followup to migrate these to be system-tests

**Review by emilyrohrbough — COMMENTED** [review]
The images being added are stale. We should log a follow up issue to update these to use Cypress 10.

File: yarn.lock [hunk 4]
```diff
@@ -13606,6 +13626,54 @@ cypress-real-events@1.6.0:
   resolved "https://registry.yarnpkg.com/cypress-real-events/-/cypress-real-events-1.6.0.tgz#277024b62a324b6937760a700e831e795c021040"
   integrity sha512-QxXm0JsQkCrb2uH+fMXNDQ5kNWTzX3OtndBafdsZmNV19j+6JuTK9n52B1YVxrDrr/qzPAojcHJc5PNoQvwp+w==
 
+cypress@*:
```
Comment (emilyrohrbough) [inline yarn.lock]: This shouldn't have been added

File: npm/grep/package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    "cy:run": "node ../../scripts/cypress.js run --config specPattern='**/unit.js'",
+    "cy:open": "node ../../scripts/cypress.js open --e2e -b electron --config specPattern='**/unit.js'"
+  },
+  "dependencies": {
+    "debug": "4.3.4",
+    "find-test-names": "^1.19.0",
+    "globby": "^11.0.4"
+  },
+  "devDependencies": {
+    "cypress": "*",
```
Comment (emilyrohrbough) [inline npm/grep/package.json]:
```suggestion
```

File: npm/grep/package.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  "name": "@cypress/grep",
+  "version": "0.0.0-development",
+  "description": "Filter tests using substring",
+  "main": "src/support",
+  "scripts": {
+    "cy:run": "node ../../scripts/cypress.js run --config specPattern='**/unit.js'",
+    "cy:open": "node ../../scripts/cypress.js open --e2e -b electron --config specPattern='**/unit.js'"
+  },
+  "dependencies": {
+    "debug": "4.3.4",
```
Comment (emilyrohrbough) [inline npm/grep/package.json]: can we open this up to this aligns with the other debug versions in the repo? or is it locked down for a reason?
```suggestion
    "debug": "^4.3.2",
```

File: npm/grep/src/plugin.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+function cypressGrepPlugin (config) {
+  if (!config || !config.env) {
+    return config
+  }
+
+  const { env } = config
+
+  if (!config.specPattern) {
+    throw new Error(
+      'Incompatible versions detected, cypress-grep 3.0.0+ requires Cypress 10.0.0+ ',
```
Comment (emilyrohrbough) [inline npm/grep/src/plugin.js]:
```suggestion
      'Incompatible versions detected, cypress-grep 3.0.0+ requires Cypress 10.0.0+',
```

**Review by emilyrohrbough — APPROVED** [review]

**Review by admah — APPROVED** [review]
Looks good. Thanks for getting this in!

**Review by jaffrepaul — APPROVED** [review]
LGTM

**Comment by emilyrohrbough** [comment]
Hey @jordanpowell88 noticed this yesterday when looking at something else, but we'll want to add this package to the list of standalone packages in the Contributing.md document: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#cypress-and-packages

**Comment by jordanpowell88** [comment]
> Hey @jordanpowell88 noticed this yesterday when looking at something else, but we'll want to add this package to the list of standalone packages in the Contributing.md document: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#cypress-and-packages

Didn't even know this list existed. Thanks @emilyrohrbough

**Pull request merged**